parmed/formats/sdf.py

```diff
@@ -0,0 +1,44 @@
+"""
+This module contains parsers for sdf file format 
+extension described at https://www.cas.org/content/chemical-suppliers/example-sdf
+"""
+from __future__ import print_function, division, absolute_import
+
+from parmed.formats.registry import FileFormatType
+from parmed.utils.io import genopen
+from parmed.utils.six import add_metaclass
+from parmed import rdkit
+
+@add_metaclass(FileFormatType)
+class SDFFile(object):
+    """ Class to read SDF file """
+
+    @staticmethod
+    def id_format(filename):
+        """ Identify the file as a SDF file format or not
+
+        Parameters
+        ----------
+        filename : str
+            Name of the file to test whether or not it is a sdf file
+
+        Returns
+        -------
+        is_fmt : bool
+            True if it is a sdf file, False otherwise
+        """
+        f = genopen(filename, 'r')
+        try:
+            for line in f:
+                if not line.strip(): continue
+                if line.startswith('M  END') or line.startswith('$$$$'):
+                    return True
+                else:
+                    continue
```

Comment:
This is going to parse the entire file if it's not an SDF file, which could substantially slow down file format detections.  Is there any way we can limit this to just looking at a few lines?